From a8e7519917082c3cadaec8b0af95d14dc3446ddd Mon Sep 17 00:00:00 2001
From: Michel Thierry <michel.thierry@intel.com>
Date: Tue, 12 Jan 2016 15:07:37 +0000
Subject: [PATCH 3/5] Revert "FROM_UPSTREAM [VPG]: drm/i915: Mark the GTT as
 uncached, not WC, for Cherryview"

This reverts commit 160f824ba7eeb52312873a3b9eb9f5b1473e3bf6.

Not only this patch was never merged upstream, but also the main issue is
already addressed in a different patch:

    Author: Chris Wilson <chris@chris-wilson.co.uk>
    Date:   Fri Oct 23 18:43:32 2015 +0100
    FROM_UPSTREAM [VPG]: drm/i915: Serialise updates to GGTT with access
    through GGTT on Braswell
    Change-Id: I9d8c950ff18034bcc473541f745fbf6544662406

Change-Id: I9315e459689db94bc095cc2be770605d4116069d
Tracked-On: https://jira01.devtools.intel.com/browse/OAM-14526
Signed-off-by: Michel Thierry <michel.thierry@intel.com>
Reviewed-on: https://android.intel.com:443/457448
---
 drivers/gpu/drm/i915/i915_gem_gtt.c | 14 +-------------
 1 file changed, 1 insertion(+), 13 deletions(-)

diff --git a/drivers/gpu/drm/i915/i915_gem_gtt.c b/drivers/gpu/drm/i915/i915_gem_gtt.c
index 2d89933..397cfd1 100644
--- a/drivers/gpu/drm/i915/i915_gem_gtt.c
+++ b/drivers/gpu/drm/i915/i915_gem_gtt.c
@@ -2494,19 +2494,7 @@ static int ggtt_probe_common(struct drm_device *dev,
 	gtt_phys_addr = pci_resource_start(dev->pdev, 0) +
 		(pci_resource_len(dev->pdev, 0) / 2);
 
-	/*
-	 * On BXT writes larger than 64 bit to the GTT pagetable range will be
-	 * dropped. For WC mappings in general we have 64 byte burst writes
-	 * when the WC buffer is flushed, so we can't use it, but have to
-	 * resort to an uncached mapping. The WC issue is easily caught by the
-	 * readback check when writing GTT PTE entries.
-	 *
-	 * Hmm, it seems that BSW also has a similar deficiency...
-	 */
-	if (IS_CHERRYVIEW(dev))
-		dev_priv->gtt.gsm = ioremap_nocache(gtt_phys_addr, gtt_size);
-	else
-		dev_priv->gtt.gsm = ioremap_wc(gtt_phys_addr, gtt_size);
+	dev_priv->gtt.gsm = ioremap_wc(gtt_phys_addr, gtt_size);
 	if (!dev_priv->gtt.gsm) {
 		DRM_ERROR("Failed to map the gtt page table\n");
 		return -ENOMEM;
-- 
1.9.1

